src/vs/workbench/contrib/chat/browser/chatWidget.ts

```diff
@@ -534,10 +534,11 @@ export class ChatWidget extends Disposable implements IChatWidget {
 		this._dynamicMessageLayoutData = { numOfMessages: numOfChatTreeItems, maxHeight };
 		this._register(this.renderer.onDidChangeItemHeight(() => this.layoutDynamicChatTreeItemMode()));
 
-		const mutableDisposable = this._register(new MutableDisposable());
+		let cts: CancellationTokenSource | undefined;
 		this._register(this.tree.onDidScroll((e) => {
-			mutableDisposable.value = dom.scheduleAtNextAnimationFrame(() => {
-				mutableDisposable.dispose();
+			cts?.cancel();
+			cts = new CancellationTokenSource();
```

Comment:
So I still need a `MutableDisposable` so that I can `_register` only one thing? If so, doesn't that make this logic more complicated than the disposable only code?

Reply:
Yeah... I think what you want is some `AnimationFrameBatcher` that is disposable and does all that.

Reply:
I'm working on a refactor that does something like this... a debounced method that will fire on the next animation frame... and the actual event handler really just collects the events. 

Ill keep playing with it. Maybe we don't need the CancellationToken here then.